### PR TITLE
ucsc_liftover: stub and nf-test added

### PR DESCRIPTION
**Pull request by kubranarci**
## PR checklist

Closes #XXX 

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - For modules:
    - [ ] `nf-core modules test <MODULE> --profile docker`
    - [ ] `nf-core modules test <MODULE> --profile singularity`
    - [ ] `nf-core modules test <MODULE> --profile conda`
  - For subworkflows:
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile docker`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile singularity`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile conda`

### CODE REVIEW

**Comment by SPPearce** [comment]
Don't forget to remove the pytest files (under` tests/nf-core/`), and the entry in the `tests/pytest_config.yml`.
If you use the tools: `nf-core modules create <TOOL> --migrate-pytest` it will do that for you automatically

**Comment by kubranarci** [comment]
> nf-core modules create <TOOL> --migrate-pytest

thank you @SPPearce I wasnt able to remember the command :D

**Comment by kubranarci** [comment]
Do you know how can I deal with  

`md5sum for empty file found:   d41d8cd98f00b204e9800998ecf8427e 
`
unlifted.bed can be empty, I should escape from this but I dont know how? 

@SPPearce @kpadm

**Comment by kubranarci** [comment]
> Do you know how can I deal with
> 
> `md5sum for empty file found: d41d8cd98f00b204e9800998ecf8427e ` unlifted.bed can be empty, I should escape from this but I dont know how?
> 
> @SPPearce @kpadm

okay this seems to be resolved :D

File: modules/nf-core/ucsc/liftover/tests/main.nf.test [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+                input[1] = Channel.of(file(params.modules_testdata_base_path + "genomics/homo_sapiens/genome/genome.chain.gz", checkIfExists: true))
+                """
+            }
+        }
+
+        then {
+            assertAll(
+                { assert process.success },
+                { assert snapshot(process.out.lifted).match() },
+                { assert file(process.out.unlifted[0][1]).exists() }
```
Comment (maxulysse) [inline modules/nf-core/ucsc/liftover/tests/main.nf.test:30]: why not snapshotting the filename?

Comment (kubranarci) [reply modules/nf-core/ucsc/liftover/tests/main.nf.test:30]: unlifted file can be empty. When I match() it, nf-core lint was complaining 

`md5sum for empty file found: d41d8cd98f00b204e9800998ecf8427e `

Comment (maxulysse) [inline modules/nf-core/ucsc/liftover/tests/main.nf.test:30]:
```suggestion
                { assert snapshot(
                    process.out.lifted,
                    file(process.out.unlifted[0][1]).name
                ).match() },
```

Comment (maxulysse) [reply modules/nf-core/ucsc/liftover/tests/main.nf.test:30]: I was thinking snapshotting the name only, not the file itself

Comment (kubranarci) [reply modules/nf-core/ucsc/liftover/tests/main.nf.test:30]: I see

**Comment by kubranarci** [comment]
@maxulysse ready to fly :D

**Review by maxulysse — APPROVED** [review]

**Pull request merged**